### PR TITLE
Automating schedule updates

### LINKED ISSUE #3: Automating schedule updates
I'm thinking to use GitHub Actions to automate the generation of schedule.json and schedule_override.json. While I have a draft Python script to generate that, I'm wondering how you usually create these files?

### PR DESCRIPTION
**Pull request by austinhuang0131**
Closes #3. The action is set to run twice a day (though maybe the run schedule should be revised).

### CODE REVIEW

**Review by steviek — APPROVED** [review]
Cool!

File: schedule-generator/index.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            data["accordionlist"][":items"].items(),
+            scheduleId,
+            title
+        ))
+    dates.sort(reverse=True)
+    r['validFrom'] = dates[0].strftime("%Y-%m-%dT00:00")
+    return r
+
+def write_json(d: dict, file: str):
+    if d == dict():
```
[the remark this answers is not in the document]
Comment (austinhuang0131) [reply schedule-generator/index.py:55]: > Is it guaranteed that we're always going to get an empty dict if something with the parsing fails?

No, if parsing fails then it'll just throw an error, so no output, we're still good

But I'll think about some procedure checks

**Comment by austinhuang0131** [comment]
btw, the CI is failing because of your branch protection rules, but it seems like there is no way to properly exempt GitHub Actions from it. We might have to put the logic in a different branch or a different repo.

**Comment by steviek** [comment]
I'll take a look into it, maybe there's some way to exempt this, or use a different branch

**Comment by steviek** [comment]
I'll just drop the rule needing a pull request for now

**Pull request merged**